airflow/www/views.py

```diff
@@ -1607,11 +1687,14 @@ def success(self):
                                               future, past, State.SUCCESS)
 
     @expose('/tree')
-    @has_dag_access(can_dag_read=True)
-    @has_access
-    @gzipped
-    @action_logging
-    def tree(self):  # pylint: disable=too-many-locals
+    @auth.has_access([
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK_INSTANCE),
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK_LOG),
```

Comment:
We should create a follow up PR to not make read task logs a requirement -- and if the permission is missing just don't do the `if task_log_reader.supports_external_link` bit.